packages/contracts/test/registry/ens-subdomain-registry.ts

```diff
@@ -0,0 +1,253 @@
+
+    // Create the registrar to be tested
+    const ENSSubdomainRegistrar = await ethers.getContractFactory(
+      'ENSSubdomainRegistrar'
+    );
+    subdomainRegistrar = await ENSSubdomainRegistrar.deploy();
+  });
+
+  describe('initialize:', () => {
+    it('initializes if the node owner has approved the subdomain registrar address', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make the signers[0] the owner
+      registerSubdomainHelper('test', '', signers[0], signers[0].address);
+
+      // Approve the subdomain registrar contract address to operate for signers[0] (who owns 'test')
+      await ens
+        .connect(signers[0])
+        .setApprovalForAll(subdomainRegistrar.address, true);
+
+      expect(
+        await subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      );
+    });
+
+    it('initializes if the subdomain registrar owns the node', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make the subdomain registrar the owner
+      registerSubdomainHelper(
+        'test',
+        '',
+        signers[0],
+        subdomainRegistrar.address
+      );
+
+      expect(
+        await subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      );
+    });
+
+    it('does not initialize if the registrar is neither the domain node owner nor an approved operator of the domain node owner', async () => {
+      await expect(
+        subdomainRegistrar
+          .connect(signers[0])
+          .initialize(dao.address, ens.address, namehash('test'))
+      ).to.be.revertedWith(
+        customError(
+          'RegistrarUnauthorized',
+          ethers.constants.AddressZero,
+          subdomainRegistrar.address
+        )
+      );
+    });
+  });
+
+  describe('registerSubnode:', () => {
+    it('registers and reassigns a subdomain if the registrar is the parent domain owner', async () => {
+      // Register the parent domain 'test' through signers[0] who owns the ENS root node ('') and make the subdomain registrar the owner
+      registerSubdomainHelper(
+        'test',
+        '',
+        signers[0],
+        subdomainRegistrar.address
+      );
+
+      // Initialize the registrar with the 'test' domain
+      subdomainRegistrar.initialize(dao.address, ens.address, namehash('test'));
+
+      // Grant signers[1] the right to register new subdomains in the subdomain registrar contract
+      await dao.grant(
+        subdomainRegistrar.address,
+        await signers[1].getAddress(),
+        REGISTER_ENS_SUBDOMAIN_ROLE
+      );
+
+      // Register the subdomain 'my.test' as signers[1] and give ownership to an address (here a DAO)
+      let tx = await subdomainRegistrar
+        .connect(signers[1])
+        .registerSubnode(labelhash('my'), dao.address);
```

Comment:
Same as before, do we want to grant "ownership" or we rather want to assign the `resolveAddress()` so that `xxx.dao.eth` points to the given DAO address?